File: BE/src/users/dto/user.dto.ts
import { User } from '../domain/user.domain';
import { ApiProperty } from '@nestjs/swagger';

export class UserDto {
  @ApiProperty({ description: 'avatarUrl' })
  avatarUrl: string;
  @ApiProperty({ description: 'userCode' })
  userCode: string;
  constructor(avatarUrl: string, userCode: string) {
    this.avatarUrl = avatarUrl;
    this.userCode = userCode;
  }
  static from(user: User) {
    return new UserDto(user.avatarUrl, user.userCode);
  }
}

File: iOS/moti/moti/Data/Sources/Data/Network/DTO/UserDTO.swift
//
//  UserDTO.swift
//  
//
//  Created by 유정주 on 11/14/23.
//

import Foundation
import Domain

struct UserTokenDTO: Codable {
    let accessToken: String?
    let refreshToken: String?
    let user: UserDTO?
}

struct UserDTO: Codable {
    let userCode: String?
    let avatarUrl: URL?
}

extension UserToken {
    init(dto: UserTokenDTO) {
        self.init(
            accessToken: dto.accessToken ?? "",
            refreshToken: dto.refreshToken,
            user: dto.user != nil ? User(dto: dto.user!) : User())
    }
}

extension User {
    init(dto: UserDTO) {
        self.init(code: dto.userCode ?? "", avatarURL: dto.avatarUrl)
    }
}
